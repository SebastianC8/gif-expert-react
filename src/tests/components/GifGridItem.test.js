import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('Pruebas <GifGridItem/>', () => {

    /**
     * Título
     */
    const title = 'Un título';
    
    /**
     * URL imagen
     */
    const url = 'https://localhost/something.jpg';

    /**
     * Componente
     */
    const wrapper = shallow(<GifGridItem title = { title } url = { url } />);

    /**
     * Se valida snapshot del component
     */
    test('Mostrar componente ok', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    /**
     * Se válida el valor del título del componente
     */
    test('Debe tener un párrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    });
    
    /**
     * Se valida el valor de los atributos de la etiqueta img del componente
     */
    test('Etiqueta img con sus respectivos atributos', () => {
       
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });

    /**
     * Se valida que un DIV del componente tenga una clase específica
     */
    test('Validar clase específica en div', () => {
       
        const div = wrapper.find('div');
        const classRequired = 'clase-test';
        const className = div.props().className;

        expect(className.includes(classRequired)).toBe(true);

    });
    

});
