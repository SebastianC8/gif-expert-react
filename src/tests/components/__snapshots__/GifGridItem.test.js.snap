// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifGridItem/> Mostrar componente ok 1`] = `
<div
  className="card clase-test"
>
  <img
    alt="Un título"
    src="https://localhost/something.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
